src/main/kotlin/com/ampnet/reportserviceeth/service/impl/TemplateDataServiceImpl.kt

```diff
@@ -59,7 +59,7 @@ class TemplateDataServiceImpl(
                 ?: throw InvalidRequestException(
                     ErrorCode.INT_UNSUPPORTED_TX, "Transaction with hash:${request.txHash} is not supported in report"
                 )
-        return SingleTransactionSummary(mappedTransaction, userWithInfo, translations)
+        return SingleTransactionSummary(mappedTransaction, userWithInfo, translations, transaction.timestamp)
```

Comment:
Why pass additional parameter `transaction.timestamp` to the class constructor when you can extract it inside a class using the same `transaction.timestamp`?